beacon-chain/sync/initial-sync/blocks_queue.go

```diff
@@ -36,6 +36,10 @@ const (
 	noRequiredPeersErrRefreshInterval = 15 * time.Second
 	// maxResetAttempts number of times stale FSM is reset, before backtracking is triggered.
 	maxResetAttempts = 4
+	// startBackSlots defines number of slots before the current head, which defines a start position
+	// of the initial machine. This allows more robustness in case of normal sync sets head to some
+	// orphaned block: in that case starting earlier and re-fetching blocks allows to reorganize chain.
+	startBackSlots = 32
```

Comment:
Is one epoch back always preferred? why not just use `SlotsPerEpoch` config?

Reply:
- It's just an arbitrary number (well I hope if head is set to detached block, the missing blocks are nearby). I made it as constant, so that if down the road we discover that we need to jump back 64 slots (say to make sure that on resync we start from definitely finalized slot, which might happen earlier than `-32` blocks) it will be just a const update away. 

If you feel strong that we should use `SlotsPerEpoch`, and inline i.e. w/o resorting to constant, I can change it -- do not feel superstrong about it.

Reply:
I dont feel strongly about this. Was mostly curious

Reply:
Any reason to use an arbitrary number instead of starting from the last finalized epoch ?

Reply:
If we start from the last finalized epoch, then this method will cancel effect of `--head-sync` i.e. on period of long non-finality we will resync from finalized instead of from position near head. 

Note: granted, we do not expect long periods of non-finality on mainnet (still quite expected on testnets), probably being too cautions here.